Homework_1.py

```diff
@@ -2,42 +2,109 @@
 
 import pandas as pd
 import plotly.express as px
+from sklearn import datasets, metrics
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
 from sklearn.preprocessing import StandardScaler
 
-data = pd.read_csv("/home/osboxes/Downloads/iris.csv")
-df1 = pd.DataFrame(data)
-df = pd.DataFrame(data)
-df.pop("class")
+# 1. loading the iris data using Pandas DataFrame
+iris = pd.DataFrame()
+x_scaled = []
+x = []
+y = []
 
-n_array = df.to_numpy()
 
-print(n_array)
+def load_database():
+    global iris
+    iris = datasets.load_iris()
+    print(iris.target_names)
+    print(iris.feature_names)
+    print(iris.data[0:5])
+    print(iris.target)
+    iris = pd.DataFrame(
+        {
+            "sepal length": iris.data[:, 0],
+            "sepal width": iris.data[:, 1],
+            "petal length": iris.data[:, 2],
+            "petal width": iris.data[:, 3],
+            "species": iris.target,
+        }
+    )
+    print(iris)
 
-mean_value = n_array.mean(axis=0)
-min_value = n_array.min(axis=0)
-max_value = n_array.max(axis=0)
 
-print("mean = ", mean_value)
-print("min = ", min_value)
-print("max = ", max_value)
+# 2. getting sum simple summary statistics (mean, min, max) using numpy
+def numpy_statistics():
+    n_array = iris.to_numpy()
+    print(n_array)
+    mean_value = n_array.mean(axis=0)
+    min_value = n_array.min(axis=0)
+    max_value = n_array.max(axis=0)
 
+    print("mean = ", mean_value)
+    print("min = ", min_value)
+    print("max = ", max_value)
 
-Histogram = px.histogram(df, x="petalLength")
-Histogram.show()
 
-scatter_plot = px.scatter(df, x="sepalLength", y="sepalWidth")
-scatter_plot.show()
+# 3. Different types of plots (Histogram, scatter plot, violin, boxplot, barchart)
 
-violin_plot = px.violin(df1, x="sepalLength", color="class")
-violin_plot.show()
 
-x = df.iloc[:, [0, 1, 2, 3]]
-x.head()
+def plot_graphs():
+    histogram = px.histogram(iris, x="petal length")
```

Comment:
iris is a local variable in this def. Please add global keyword to include the global variable "iris"